server/consumer.go

```diff
@@ -297,6 +303,10 @@ func setConsumerConfigDefaults(config *ConsumerConfig) {
 	if config.MaxDeliver == 0 {
 		config.MaxDeliver = -1
 	}
+	// If BackOff was specified that will override the AckWait and the MaxDeliver.
+	if len(config.BackOff) > 0 {
+		config.AckWait = config.BackOff[0]
```

Comment:
Does it mean that consumers must be aware that they must acknowledge before the backoff duration at index 0? As anything slower than that will represent the message will already be redelivered for the 2nd attempt.

If I'm not wrong, the `NAK` delay is applied when it reaches the server again, adding to the time spent already on the application code. For me is easier to think of backoff as an addition to the AckWait as the redeliver will start counting from that moment plus the backoff.

Of course, application side can choose another backoff to follow that reasoning, but when I get this, I get surprised about it.